File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlSyntaxException.java
package org.cqframework.cql.cql2elm;

import org.cqframework.cql.elm.tracking.TrackBack;

/**
 * Created by Bryn on 3/27/2017.
 */
public class CqlSyntaxException extends CqlCompilerException {
    public CqlSyntaxException(String message) {
        super(message);
    }

    public CqlSyntaxException(String message, ErrorSeverity severity) {
        super(message, severity);
    }

    public CqlSyntaxException(String message, Throwable cause) {
        super(message, cause);
    }

    public CqlSyntaxException(String message, ErrorSeverity severity, Throwable cause) {
        super(message, severity, cause);
    }

    public CqlSyntaxException(String message, TrackBack locator) {
        super(message, locator);
    }

    public CqlSyntaxException(String message, ErrorSeverity severity, TrackBack locator) {
        super(message, severity, locator);
    }

    public CqlSyntaxException(String message, TrackBack locator, Throwable cause) {
        super(message, locator, cause);
    }

    public CqlSyntaxException(String message, ErrorSeverity severity, TrackBack locator, Throwable cause) {
        super(message, severity, locator, cause);
    }
}

File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlSyntaxException.kt
package org.cqframework.cql.cql2elm

import org.cqframework.cql.elm.tracking.TrackBack

/** Created by Bryn on 3/27/2017. */
class CqlSyntaxException : CqlCompilerException {
    constructor(message: String?) : super(message)

    constructor(message: String?, severity: ErrorSeverity) : super(message, severity)

    constructor(message: String?, cause: Throwable?) : super(message, cause)

    constructor(
        message: String?,
        severity: ErrorSeverity,
        cause: Throwable?
    ) : super(message, severity, cause)

    constructor(message: String?, locator: TrackBack?) : super(message, locator)

    constructor(
        message: String?,
        severity: ErrorSeverity,
        locator: TrackBack?
    ) : super(message, severity, locator)

    constructor(
        message: String?,
        locator: TrackBack?,
        cause: Throwable?
    ) : super(message, locator, cause)

    constructor(
        message: String?,
        severity: ErrorSeverity,
        locator: TrackBack?,
        cause: Throwable?
    ) : super(message, severity, locator, cause)
}
